tests/test_salesforce_lookback_window.py

```diff
@@ -0,0 +1,105 @@
+        return {
+            'Account'
+        }
+
+    def run_test(self):
+        # create connection
+        conn_id = connections.ensure_connection(self)
+        # create state file
+        state = {
+            'bookmarks':{
+                'Account': {
+                    'SystemModstamp': '2021-11-10T00:00:00.000000Z'
+                }
+            }
+        }
+        # set state file to run in sync mode
+        menagerie.set_state(conn_id, state)
+
+        # run in check mode
+        found_catalogs = self.run_and_verify_check_mode(conn_id)
+
+        # select certain catalogs
+        expected_streams = self.expected_sync_streams()
+        catalog_entries = [catalog for catalog in found_catalogs
+                            if catalog.get('tap_stream_id') in expected_streams]
+
+        # stream and field selection
+        self.select_all_streams_and_fields(conn_id, catalog_entries)
+
+        # run sync
+        self.run_and_verify_sync(conn_id)
+
+        # get synced records
+        sync_records = runner.get_records_from_target_output()
+
+        # get replication keys
+        expected_replication_keys = self.expected_replication_keys()
+
+        # get bookmark ie. date from which the sync started
+        bookmark = state.get('bookmarks').get('Account').get('SystemModstamp')
+        # calculate the simulated bookmark by subtracting lookback window seconds
+        bookmark_with_lookback_window = self.timedelta_formatted(bookmark, format=self.BOOKMARK_COMPARISON_FORMAT, seconds=-self.get_properties()['lookback_window'])
+
+        for stream in expected_streams:
+            with self.subTest(stream=stream):
+
+                # get replication key for stream
+                replication_key = list(expected_replication_keys[stream])[0]
+
+                # get records
+                records = [record.get('data') for record in sync_records.get(stream).get('messages')
+                           if record.get('action') == 'upsert']
+
+                # check for the record if it is between lookback date and bookmark
+                is_between = False
+
+                for record in records:
+                    replication_key_value = record.get(replication_key)
+
+                    # Verify the sync records respect the (simulated) bookmark value
+                    self.assertGreaterEqual(self.parse_date(replication_key_value), self.parse_date(bookmark_with_lookback_window),
+                                            msg='The record does not respect the lookback window.')
+
+                    # verify if the record's bookmark value is between bookmark and (simulated) bookmark value
+                    if self.parse_date(bookmark_with_lookback_window) <= self.parse_date(replication_key_value) < self.parse_date(bookmark):
+                        is_between = True
+
+                    self.assertTrue(is_between, msg='No record found between bookmark and lookback date.')
```

Comment:
Ideally, if we are not using any lookback then we will get all data greater than the bookmark. Here, we are using lookback so, above assertion to verify we getting at least some data between `bookmark - lookback` and `bookmark`

Reply:
I think Andy's point is that we could assert on just the first record. But for that assertion to be valid, we'd need to also assert that records are returned in order of ascending replication key value. Which I think is a better approach, but both are valid. 
Also it's always preferable to avoid assertTrue and assertFalse. They give less information than the comparison methods, even with the additional error msg.

Reply:
@kspeer825 @luandy64 Added assertion to verify we got records in ascending order and also updated the existing `is_between` assertion to verify that the 1st record is between the lookback date and the initial bookmark.